File: tests/test_data/Fields.java
package test_data;

public class Fields {
    public boolean t_boolean = true;
    public int t_int = 0;
    public long t_long = 100L;
    public float t_float = 1.0f;
    public double t_double = 2.0d;

    public static boolean t_boolean() {
        Fields a = new Fields();
        return a.t_boolean;
    }

    public static int t_int() {
        Fields a = new Fields();
        return a.t_int;
    }

    public static long t_long() {
        Fields a = new Fields();
        return a.t_long;
    }

    public static float t_float() {
        Fields a = new Fields();
        return a.t_float;
    }

    public static double t_double() {
        Fields a = new Fields();
        return a.t_double;
    }
}

File: tests/fields.rs
extern crate rjvm;

use rjvm::vm::data_type::Value::{Double, Float, Int, Long};

#[path = "utils.rs"]
mod java;

#[test]
fn test_static_boolean_field() {
    assert_eq!(java::run_method("test_data/Fields", "s_boolean"), Int(0));
}

#[test]
fn test_static_int_field() {
    assert_eq!(java::run_method("test_data/Fields", "s_int"), Int(100));
}

#[test]
fn test_static_long_field() {
    assert_eq!(java::run_method("test_data/Fields", "s_long"), Long(200));
}

#[test]
fn test_static_float_field() {
    assert_eq!(
        java::run_method("test_data/Fields", "s_float"),
        Float(300.0)
    );
}

#[test]
fn test_static_double_field() {
    assert_eq!(
        java::run_method("test_data/Fields", "s_double"),
        Double(400.0)
    );
}

#[test]
fn test_static_other() {
    assert_eq!(java::run_method("test_data/Fields", "other"), Int(10));
}

/*
#[test]
fn test_boolean_field() {
    assert_eq!(run_method("t_boolean"), Boolean(true));
}*/// TODO

#[test]
fn test_int_field() {
    assert_eq!(java::run_method("test_data/Fields", "t_int"), Int(0));
}

#[test]
fn test_long_field() {
    assert_eq!(java::run_method("test_data/Fields", "t_long"), Long(100));
}

#[test]
fn test_float_field() {
    assert_eq!(java::run_method("test_data/Fields", "t_float"), Float(1.0));
}

#[test]
fn test_double_field() {
    assert_eq!(
        java::run_method("test_data/Fields", "t_double"),
        Double(2.0)
    );
}
